File: 1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442. Count Triplets That Can Form Two Arrays of Equal XOR.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        n = len(arr)
        prefix_xor = [0] * (n + 1)
        for i in range(n):
            prefix_xor[i + 1] = prefix_xor[i] ^ arr[i]
        
        count = 0
        for i in range(n - 1):
            c = arr[i]
            for k in range(i + 1, n):
                a = prefix_xor[k] ^ prefix_xor[i]
                if a == arr[k]:
                    b = prefix_xor[k + 1] ^ prefix_xor[k]
                    if a == b:
                        count += k - i
        
        return count

File: 1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442. Count Triplets That Can Form Two Arrays of Equal XOR.ts
function countTriplets(arr) {
	const n = arr.length;
	const prefix_xor = new Array(n + 1).fill(0);
	for (let i = 0; i < n; i++) {
		prefix_xor[i + 1] = prefix_xor[i] ^ arr[i];
	}

	let count = 0;
	for (let i = 0; i < n - 1; i++) {
		const c = arr[i];
		for (let k = i + 1; k < n; k++) {
			const a = prefix_xor[k] ^ prefix_xor[i];
			if (a === arr[k]) {
				const b = prefix_xor[k + 1] ^ prefix_xor[k];
				if (a === b) {
					count += k - i;
				}
			}
		}
	}

	return count;
}
